pvlib/ivtools/params.py

```diff
@@ -0,0 +1,114 @@
+"""
+The ``params`` module contains classes and functions to extract parameters
+(e.g. Isc and Voc) from current-voltage curves.
+"""
+import pandas as pd
+import numpy as np
+from numpy.polynomial.polynomial import Polynomial as Poly
+
+
+def astm_e1036(v, i, imax_limits=(0.75, 1.15), vmax_limits=(0.75, 1.15),
+               voc_points=3, isc_points=3):
+    '''
+    Extract photovoltaic IV parameters according to ASTM E1036. Assumes the
+    curve is in the first quadrant
+
+    Parameters
+    ----------
+    v : array-like
+        Voltage points
+    i : array-like
+        Current points
+    imax_limits : tuple
+        Two-element tuple (low, high) specifying the fraction of estimated
+        Imp within which to fit a polynomial for max power calculation
+    vmax_limits : tuple
+        Two-element tuple (low, high) specifying the fraction of estimated
+        Vmp within which to fit a polynomial for max power calculation
+    voc_points : int
+        the number of points near open circuit to use for linear fit
+        and Voc calculation
+    isc_points : int
+        the number of points near short circuit to use for linear fit and
+        Isc calculation
+
+    Returns
+    -------
+    dict
+        Calculated IV parameters
+
+    Adapted from https://github.com/NREL/iv_params
+    Copyright (c) 2022, Alliance for Sustainable Energy, LLC
+    All rights reserved.
+    '''
+
+    df = pd.DataFrame()
```

Comment:
I think everything could all be calculated with numpy only, if that is of interest.

Reply:
Interesting suggestion. I don't think I will have time to tackle it, but if you feel it makes a sufficient improvement please go for it. Tests are in place now so it should be easy to verify that it has worked.